addons/hr_holidays/models/hr_leave.py

```diff
@@ -1412,8 +1413,12 @@ def _action_user_cancel(self, reason):
         if not self.can_cancel:
             raise ValidationError(_('This time off cannot be canceled.'))
 
+        self._force_cancel(reason, 'mail.mt_note')
+
+    def _force_cancel(self, reason, msg_subtype='mail.mt_comment'):
         self.message_post(
-            body=_('The time off has been canceled: %s', reason)
```

Comment:
You can make this method work in batch and call it once on all the leaves to be canceled